Create timer with clock

The [rcl](https://github.com/ros2/rcl) has landed a [patch](https://github.com/ros2/rcl/commit/b324f9de0de49431557bcfb398ef3ad01bf46f86) last weekend, which will initialize `timer` with `clock`.

The current codebase of `rclnodejs` will report compile error based on the latest ros2 `master`, see below:

``` bash
In file included from ../src/rcl_bindings.cpp:33:0:
../src/rcl_bindings.cpp: In function ‘Nan::NAN_METHOD_RETURN_TYPE rclnodejs::CreateTimer(Nan::NAN_METHOD_ARGS_TYPE)’:
/home/minggang/ros2_ws/install/include/rcutils/time.h:32:54: error: invalid conversion from ‘int64_t {aka long int}’ to ‘rcl_clock_t*’ [-fpermissive]
 #define RCUTILS_MS_TO_NS(milliseconds) (milliseconds * (1000 * 1000))
                                                      ^
../src/macros.hpp:20:16: note: in definition of macro ‘CHECK_OP_AND_THROW_ERROR_IF_NOT_TRUE’
     if (lhs op rhs) { \
                ^
../src/rcl_bindings.cpp:72:3: note: in expansion of macro ‘THROW_ERROR_IF_NOT_EQUAL’
   THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK,
   ^
/home/minggang/ros2_ws/install/include/rcl/time.h:32:22: note: in expansion of macro ‘RCUTILS_MS_TO_NS’
 #define RCL_MS_TO_NS RCUTILS_MS_TO_NS
                      ^
../src/rcl_bindings.cpp:73:50: note: in expansion of macro ‘RCL_MS_TO_NS’
                            rcl_timer_init(timer, RCL_MS_TO_NS(period_ms),
                                                  ^
../src/rcl_bindings.cpp:74:79: error: cannot convert ‘std::nullptr_t’ to ‘int64_t {aka long int}’ for argument ‘3’ to ‘rcl_ret_t rcl_timer_init(rcl_timer_t*, rcl_clock_t*, int64_t, rcl_timer_callback_t, rcl_allocator_t)’
                                           nullptr, rcl_get_default_allocator()),
                                                                               ^
../src/macros.hpp:20:16: note: in definition of macro ‘CHECK_OP_AND_THROW_ERROR_IF_NOT_TRUE’
     if (lhs op rhs) { \
                ^
../src/rcl_bindings.cpp:72:3: note: in expansion of macro ‘THROW_ERROR_IF_NOT_EQUAL’
   THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK,
   ^
rclnodejs.target.mk:118: recipe for target 'Release/obj.target/rclnodejs/src/rcl_bindings.o' failed
make: *** [Release/obj.target/rclnodejs/src/rcl_bindings.o] Error 1
```

We are going to change the implementation to adapt to the latest code to eliminate the error.

--- a/src/rcl_bindings.cpp
+++ b/src/rcl_bindings.cpp
@@ -69,13 +69,26 @@ NAN_METHOD(CreateTimer) {
   rcl_timer_t* timer =
       reinterpret_cast<rcl_timer_t*>(malloc(sizeof(rcl_timer_t)));
   *timer = rcl_get_zero_initialized_timer();
+  rcl_clock_t* clock =
+      reinterpret_cast<rcl_clock_t*>(malloc(sizeof(rcl_clock_t)));
+  rcl_allocator_t allocator = rcl_get_default_allocator();
   THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK,
-                           rcl_timer_init(timer, RCL_MS_TO_NS(period_ms),
+                           rcl_clock_init(RCL_STEADY_TIME,
+                                          clock,
+                                          &allocator),
+                           rcl_get_error_string_safe());
+  THROW_ERROR_IF_NOT_EQUAL(RCL_RET_OK,
+                           rcl_timer_init(timer, clock, RCL_MS_TO_NS(period_ms),
                                           nullptr, rcl_get_default_allocator()),
                            rcl_get_error_string_safe());
 
   auto js_obj = RclHandle::NewInstance(
-      timer, nullptr, [timer] { return rcl_timer_fini(timer); });
+      timer, nullptr, [timer, clock] {
+        rcl_ret_t ret_clock = rcl_clock_fini(clock);
+        free(clock);
+        rcl_ret_t ret_timer = rcl_timer_fini(timer);
+        return ret_clock || ret_timer;
+      });
   info.GetReturnValue().Set(js_obj);
 }
 

